### PR TITLE
feat(impress): Allow inserting multimedia

### PR DESCRIPTION
**Pull request by Minion3665**
Previously we could play videos/audio inserted from LibreOffice, but had no way to insert them from online. This commit adds a new "Multimedia" button to the insert menu of impress, which supports inserting different audio/video formats.

This relies on https://gerrit.libreoffice.org/c/core/+/174740 (or to be running inproc in some environments where the sandboxing is weaker) as otherwise it is unable to send the size of the inserted video to core, which will cause core to fail to insert the new element.

Change-Id: I881818277edb69ee1e98ba1bd3454e8d29593889

* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] I have run `make prettier-write` and formatted the code.
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Comment by Minion3665** [comment]
```diff
[skyler@canvasback:~/Code/collabora/jj/online]$ jj interdiff --from b001a5636ca82fb51fa2da34787510bce508788b --to @-- --git
diff --git a/kit/ChildSession.cpp b/kit/ChildSession.cpp
index 2857bce20f..04423bba6e 100644
--- a/kit/ChildSession.cpp
+++ b/kit/ChildSession.cpp
@@ -1716,18 +1716,12 @@
                     "\"value\":\"false\""
                 "},"
                 "\"Size\":{"
-                    "\"type\":\"[]com.sun.star.beans.PropertyValue\","
+                    "\"type\":\"com.sun.star.awt.Size\","
                     "\"value\":{"
                         // We don't know the real size, but the core we have can't calculate it for us either due to a lack of gstreamer. Use the default...
                         // TODO: someday, maybe we could send this from online, perhaps calculating it by using a video element, if we're inserting a video?
-                        "\"Width\":{"
-                            "\"type\":\"long\","
-                            "\"value\":0"
-                        "},"
-                        "\"Height\":{"
-                            "\"type\":\"long\","
-                            "\"value\":0"
-                        "}"
+                        "\"Width\":0,"
+                        "\"Height\":0"
                     "}"
                 "}"
             "}";

[skyler@canvasback:~/Code/collabora/jj/online]$ jj interdiff --from 46b00949b7419de58f58164f33df03e9323b2fc9 --to @- --git

[skyler@canvasback:~/Code/collabora/jj/online]$ 
```

Rebased at the same time so GitHub will not be able to sensibly show you the changes between forcepushes ... here's the interdiffs

**Comment by Minion3665** [comment]
possibly nice to see this one (which is from the start to now) too
```diff
[skyler@canvasback:~/Code/collabora/jj/online]$ jj interdiff --from b001a5636ca82fb51fa2da34787510bce508788b --to bf02f18bb548 --git
diff --git a/kit/ChildSession.cpp b/kit/ChildSession.cpp
index 2857bce20f..365b8da610 100644
--- a/kit/ChildSession.cpp
+++ b/kit/ChildSession.cpp
@@ -1716,17 +1716,20 @@
                     "\"value\":\"false\""
                 "},"
                 "\"Size\":{"
-                    "\"type\":\"[]com.sun.star.beans.PropertyValue\","
+                    "\"type\":\"any\","
                     "\"value\":{"
-                        // We don't know the real size, but the core we have can't calculate it for us either due to a lack of gstreamer. Use the default...
-                        // TODO: someday, maybe we could send this from online, perhaps calculating it by using a video element, if we're inserting a video?
-                        "\"Width\":{"
-                            "\"type\":\"long\","
-                            "\"value\":0"
-                        "},"
-                        "\"Height\":{"
-                            "\"type\":\"long\","
-                            "\"value\":0"
+                        "\"type\":\"com.sun.star.awt.Size\","
+                        "\"value\":{"
+                            // We don't know the real size, but the core we have can't calculate it for us either due to a lack of gstreamer. Use the default...
+                            // TODO: someday, maybe we could send this from online, perhaps calculating it by using a video element, if we're inserting a video?
+                            "\"Width\":{"
+                                "\"type\":\"long\","
+                                "\"value\":0"
+                            "},"
+                            "\"Height\":{"
+                                "\"type\":\"long\","
+                                "\"value\":0"
+                            "}"
                         "}"
                     "}"
                 "}"
```

**Review by eszkadev — APPROVED** [review]
Works, let's get this in, thanks!

**Comment by Minion3665** [comment]
awesome! will wait for CI to pass then merge...

**Comment by eszkadev** [comment]
unrelated writer invalidation tet fail, restarted cypress

File: browser/src/unocommands.js [hunk 1]
```diff
@@ -276,6 +276,7 @@ var unoCommandsArray = {
 	'InsertFootnote':{text:{context:_('Insert Footnote'),menu:_('~Footnote'),},},
 	'InsertFrame':{text:{context:_('Insert Frame'),menu:_('F~rame...'),},},
 	'InsertGraphic':{global:{context:_('Insert Image...'),menu:_('~Image...'),},},
+	'InsertAVMedia':{global:{context:_('Insert Multimedia...'),menu:_('Multimedia...'),},},
```
Comment (timar) [inline browser/src/unocommands.js:279]: @Minion3665:
Manual update of this file is prohibited. In core co-24.04 branch the strings are different, as it turned out when I updated UNO command translations.
```
'InsertAVMedia':{global:{context:_('Insert Audio or Video'),menu:_('Media'),},},
```
But then, the "Local Multimedia" menu item will differ. This has to be sorted out in a follow-up patch.

**Pull request merged**